src/_gettsim/transfers/arbeitsl_geld_2/kindergelduebertrag.py

```diff
@@ -125,39 +126,60 @@ def _diff_kindergeld_kindbedarf_m(  # noqa: PLR0913
         See :func:`arbeitsl_geld_2_regelbedarf_m_bg`.
     arbeitsl_geld_2_nettoeink_nach_abzug_freibetrag_m
         See :func:`_arbeitsl_geld_2
-    wohngeld_anspruchshöhe_m_bg
-        See :func:`wohngeld_anspruchshöhe_m_bg`.
+    wohngeld_m_wthh
+        See :func:`wohngeld_m_wthh`.
+    anz_personen_wthh
+        See :func:`anz_personen_wthh`.
     kindergeld_zur_bedarfsdeckung_m
         See :func:`kindergeld_zur_bedarfsdeckung_m`.
     kind_unterh_erhalt_m
         See :func:`kind_unterh_erhalt_m`.
     unterhaltsvors_m
         See :func:`unterhaltsvors_m`.
-    eigenbedarf_gedeckt
-        See :func:`eigenbedarf_gedeckt`.
+    _in_anderer_bedarfsgemeinschaft_als_kindergeldempfänger
+        See :func:`_in_anderer_bedarfsgemeinschaft_als_kindergeldempfänger`.
 
     Returns
     -------
 
     """
-    # TODO (@MImmesberger): Remove `eigenbedarf_gedeckt` conditions once
-    # Bedarfsgemeinschaft is fully endogenous. This is a temporary fix. Without it,
-    # Kindergeld would be counted twice as income of the Bedarfsgemeinschaft (one time
-    # the full amount for the child and one time the Kindergeldübertrag for the parent -
-    # because the child doesn't drop out of Bedarfsgemeinschaft endogenously).
-    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/758
     fehlbetrag = max(
         arbeitsl_geld_2_regelbedarf_m_bg
-        - wohngeld_anspruchshöhe_m_bg
+        - wohngeld_m_wthh / anz_personen_wthh
         - arbeitsl_geld_2_nettoeink_nach_abzug_freibetrag_m
         - kind_unterh_erhalt_m
         - unterhaltsvors_m,
         0.0,
     )
     # Bedarf not covered or same Bedarfsgemeinschaft as parents
-    if not eigenbedarf_gedeckt or fehlbetrag > kindergeld_zur_bedarfsdeckung_m:
+    if (
+        not _in_anderer_bedarfsgemeinschaft_als_kindergeldempfänger
+        or fehlbetrag > kindergeld_zur_bedarfsdeckung_m
+    ):
         out = 0.0
     # Bedarf is covered
     else:
         out = kindergeld_zur_bedarfsdeckung_m - fehlbetrag
     return out
+
+
+@policy_info(skip_vectorization=True)
+def _in_anderer_bedarfsgemeinschaft_als_kindergeldempfänger(
+    p_id: numpy.ndarray[int],
+    p_id_kindergeld_empf: numpy.ndarray[int],
+    bg_id: numpy.ndarray[int],
+) -> numpy.ndarray[bool]:
+    """True if the person is in a different Bedarfsgemeinschaft than the
```

Comment:
I guess you are primarily thinking about cases where `fg_id` is the same here? Maybe make that explicit in the docstring.

Reply:
Yes, I think so. I'm actually not sure what happens with Kindergeldübertrag if both live in different households (and thus are in different fg's).